server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1736,19 +1780,6 @@ private static boolean ensureAliasOrIndexExists(Context context, String name) {
             return true;
         }
 
-        private static void validateAliasOrIndex(ExpressionList.Expression expression) {
```

Comment:
No point in having a single use method like that not inside the loop body, it's just harder on the compiler a well as the reader of the code.